### PR TITLE
New Order Flags & toPreview()

### PR DESCRIPTION
**Pull request by f3rno**
Adds the new order flags, and a `toPreview()` method for the UI

```js
Order.flags = {
  OCO: 2**14,      // 16384
  POSTONLY: 2**12, // 4096
  HIDDEN: 2**6,    // 64
  NO_VR: 2**19     // 524288
}
```

Flags can be specified with a key in the constructor, like `new Order({ hidden: true ... })` for hidden, postonly, and OCO. The OCO flag uses the value of `price_aux_limit` for the OCO stop price. Along with support for flags, I've added helpers for accessing/modifying them via `setHidden`, `isHidden`, etc.

The `flags` field defaults to 0 in the `Order` constructor if not otherwise set.

As per Robert's suggestion, I've swapped `isNaN` out for `Number.isNaN` & removed `String(..)` in a few places.

### CODE REVIEW

File: lib/models/order.js [hunk 1]
```diff
@@ -201,18 +279,25 @@ class Order extends Model {
...
-      price_aux_limit: `${this.priceAuxLimit || ''}`,
       price: `${this.price}`,
       amount: `${this.amount}`,
-      hidden: this.hidden ? 1 : 0,
-      postonly: this.postonly ? 1 : 0
+      flags: this.flags || 0
+    }
+
+    if (!Number.isNaN(this.priceTrailing)) {
+      data.price_trailing = this.priceTrailing + ''
```
Comment (robertkowalski) [inline lib/models/order.js]: sorry, had to test again.

```js
const foo = undefined

Number.isNaN(foo) // false
```

you probably want to cast:

```js
const foo = undefined
+foo // NaN

// so:
Number.isNaN(+foo) // true


```

Comment (robertkowalski) [reply lib/models/order.js]: currently we are producing order like

```
{ gid: null,
  cid: 1519129056809,
  symbol: 'tBTCUSD',
  type: 'EXCHANGE LIMIT',
  price: '100',
  amount: '1',
  flags: 0,
  price_trailing: 'undefined',
  price_aux_limit: 'undefined' }
```

File: lib/models/order.js [hunk 2]
```diff
@@ -201,18 +217,33 @@ class Order extends Model {
...
-      price_trailing: `${this.priceTrailing || ''}`,
-      price_aux_limit: `${this.priceAuxLimit || ''}`,
       price: `${this.price}`,
       amount: `${this.amount}`,
-      hidden: this.hidden ? 1 : 0,
-      postonly: this.postonly ? 1 : 0
+      flags: 0
     }
+
+    if (!isNaN(this.priceTrailing)) {
```
[the remark this answers is not in the document]
Comment (f3rno) [reply lib/models/order.js]: Updated

File: lib/models/order.js [hunk 3]
```diff
@@ -201,18 +272,25 @@ class Order extends Model {
...
       gid: this.gid,
-      cid: isNaN(this.cid) ? lastCID++ : this.cid,
+      cid: Number.isNaN(+this.cid) ? lastCID++ : this.cid,
       symbol: this.symbol,
       type: this.type,
-      price_trailing: `${this.priceTrailing || ''}`,
-      price_aux_limit: `${this.priceAuxLimit || ''}`,
       price: `${this.price}`,
       amount: `${this.amount}`,
-      hidden: this.hidden ? 1 : 0,
```
Comment (robertkowalski) [inline lib/models/order.js]: i think this is the issue i mentioned in chat

**Pull request merged**